src/cmd/services/m3coordinator/downsample/options.go

```diff
@@ -978,6 +987,14 @@ func (cfg Configuration) newAggregator(o DownsamplerOptions) (agg, error) {
 	}, nil
 }
 
+func initStoreNamespaces(store kv.Store, nsKey string) error {
+	_, err := store.SetIfNotExists(nsKey, &rulepb.Namespaces{})
```

Comment:
How safe is it to "set if not exist" while multiple writes might be happening to the store? It's probably a corner case since this logic can cause problems only with external uninitialized Etcd.

Reply:
You'll get back a `kv.ErrAlreadyExists` if there is a race. Etcd guarantees consistency since it each transaction uses raft consensus, so only one will win the race and the others will get back `kv.ErrAlreadyExists`.